### ISSUE TITLE
Compatibility with the new javascript module api

### ISSUE THREAD

**Issue by bartocc** [typed-ember/ember-cli-typescript#39]
RFC: https://github.com/emberjs/rfcs/blob/master/text/0176-javascript-module-api.md

```typescript
import Component from '@ember/component';
```

generates an error : `[ts] Cannot find module '@ember/component'.`

This import is possible thanks to https://github.com/ember-cli/ember-cli-shims/pull/92 available in https://github.com/ember-cli/ember-cli-shims/releases/tag/v1.2.0-beta.2.

How can we make typescript aware of those new modules ?

**Comment by chriskrycho** [comment]
@bartocc I believe what we'll need to do is update the typings in the Definitely Typed repo for that. We'll need to come up with a strategy for defining the imports at *both* locations in some way or another, given that we're certain to have some users on the old and some on the new import paths. 🤔

**Comment by bartocc** [comment]
@chriskrycho that's roughly what I thought. But I am not familiar with the Definitely Typed repo and the way @types/ember is generated. Do you know who might know more about this ?

**Comment by chriskrycho** [comment]
Unfortunately, it's generated *by hand*. It's a matter of forking the repo and then updating [this file](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ember/index.d.ts). I started on some tweaks a few weeks ago (before I went on vacation) but ran into a number of issues there around extracting the `RSVP` bits to reference [the dedicated RSVP typings](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/rsvp/index.d.ts). My goal is to get that done first, then put together the plan for updating the typings in general. I need to work through and identify *what needs to be done* and then I'll try to write up an issue in terms of things people can tackle.
